### PR TITLE
use gnns status if available

### PR DESCRIPTION
**Pull request by bobzilladev**

### CODE REVIEW

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/MainActivity.java [hunk 1]
```diff
@@ -2126,6 +2153,15 @@ private void internalSetLocationUpdates(final long updateIntervalMillis, final f
         }
     }
 
+    public static <A> String join(final String delimiter, final Iterable<A> iterable) {
```
Comment (rksh) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/MainActivity.java]: i can't fault you for doing this. android is slow. my only caution is null checking.

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/listener/GPSListener.java [hunk 2]
```diff
@@ -358,16 +371,61 @@ private boolean horribleGps(final Location location) {
...
         }
         return satCount;
     }
 
+    public Map<String, Integer> getConstellations() {
+        final Map<String, Integer> cons = new TreeMap<>();
+        if (gnssStatus != null && Build.VERSION.SDK_INT >= 24) {
+            for ( int i = 0; i < gnssStatus.getSatelliteCount(); i++ ) {
+                if ( gnssStatus.usedInFix(i) ) {
+                    final String key = constellationToString(gnssStatus.getConstellationType(i));
```
Comment (rksh) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/listener/GPSListener.java]: what do these look like x-lang?

Comment (bobzilladev) [reply wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/listener/GPSListener.java]: It's a static proper noun, so same in all languages.

**Comment by bobzilladev** [comment]
Pushed null checks for join, and a fix to make it buildable in non-q-land.

**Pull request merged**